python/ray/serve/storage/tests/test_kv_store.py

```diff
@@ -0,0 +1,98 @@
+import os
+
+import pytest
+from ray.serve.storage.kv_store import (RayInternalKVStore, RayLocalKVStore,
+                                        RayS3KVStore)
+from ray.serve.tests.conftest import (  # noqa: F401 imported but unused
+    serve_instance, shared_serve_instance)
+
+
+def test_ray_internal_kv(serve_instance):  # noqa: F811
+    with pytest.raises(TypeError):
+        RayInternalKVStore(namespace=1)
+        RayInternalKVStore(namespace=b"")
+
+    kv = RayInternalKVStore()
+
+    with pytest.raises(TypeError):
+        kv.put(1, b"1")
+    with pytest.raises(TypeError):
+        kv.put("1", 1)
+    with pytest.raises(TypeError):
+        kv.put("1", "1")
+
+    kv.put("1", b"2")
+    assert kv.get("1") == b"2"
+    kv.put("2", b"4")
+    assert kv.get("2") == b"4"
+    kv.put("1", b"3")
+    assert kv.get("1") == b"3"
+    assert kv.get("2") == b"4"
+
+
+def test_ray_internal_kv_collisions(serve_instance):  # noqa: F811
+    kv1 = RayInternalKVStore()
+    kv1.put("1", b"1")
+    assert kv1.get("1") == b"1"
+
+    kv2 = RayInternalKVStore("namespace")
+
+    assert kv2.get("1") is None
+
+    kv2.put("1", b"-1")
+    assert kv2.get("1") == b"-1"
+    assert kv1.get("1") == b"1"
+
+
+def _test_operations(kv_store):
+    # Trival get & put
+    kv_store.put("1", b"1")
+    assert kv_store.get("1") == b"1"
+    kv_store.put("2", b"2")
+    assert kv_store.get("1") == b"1"
+    assert kv_store.get("2") == b"2"
+
+    # Overwrite same key
+    kv_store.put("1", b"-1")
+    assert kv_store.get("1") == b"-1"
+
+    # Get non-existing key
+    assert kv_store.get("3") is None
+
+    # Delete existing key
+    kv_store.delete("1")
+    kv_store.delete("2")
+    assert kv_store.get("1") is None
+    assert kv_store.get("2") is None
+
+    # Delete non-existing key
+    kv_store.delete("3")
+
+
+def test_ray_serve_external_kv_local_disk():
+    kv_store = RayLocalKVStore("namespace", "test_kv_store.db")
+
+    _test_operations(kv_store)
+
+    if os.path.exists("/tmp/ray_serve_checkpoint_key.txt"):
+        os.remove("/tmp/ray_serve_checkpoint_key.txt")
+    if os.path.exists("/tmp/ray_serve_checkpoint_val.txt"):
+        os.remove("/tmp/ray_serve_checkpoint_val.txt")
```

Comment:
are these still relevant now that we're using sqlite3?

Reply:
nope, should remove :)